### PR TITLE
Add Azure cloud provider support document

### PR DESCRIPTION
**Pull request by c3y1huang**
# Describe your changes
The skuba added support for in-tree Azure Cloud Provider.

# Related Issues / Projects

Relates to:
* [Support Azure in-tree cloud provider #1582](https://github.com/SUSE/avant-garde/issues/1582)
* [Skuba support In-tree Azure Cloud Provider #1148](https://github.com/SUSE/skuba/pull/1148)
* [New test-cases for Azure as cloud provider #44](https://github.com/SUSE/caasp-test-cases/pull/44)
* [add documents for azure terraform #804](https://github.com/SUSE/doc-caasp/pull/804)

Signed-off-by: Chin-Ya Huang <chin-ya.huang@suse.com>

### CODE REVIEW

File: adoc/admin-storage-vsphere.adoc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 = vSphere Storage
-The vSphere cloud provider can be enabled with {productname} to allow Kubernetes pods to use VMWare vSphere Virtual Machine Disk (VMDK) volumes as persistent storage.
+The vSphere cloud provider can be enabled with {productname} to allow Kubernetes pods to use VMWare vSphere Virtual Machine Disk (VMDK) volumes as data volumes.
```
Comment (innobead) [inline adoc/admin-storage-vsphere.adoc:2]:
```suggestion
The vSphere cloud provider can be enabled with {productname} to allow Kubernetes pods to use VMWare vSphere Virtual Machine Disk (VMDK) volumes as persistent volumes.
```

File: adoc/deployment-bootstrap.adoc [hunk 2]
```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
 You can use the "private dns" values provided by the {tf} output.
 ====
 
+====== Azure In-tree CPI (VCP)
```
Comment (jenting) [inline adoc/deployment-bootstrap.adoc:293]: It's not VCP (VMWare Cloud Provider), right?

File: adoc/admin-storage-azure.adoc [hunk 3]
```diff
@@ -0,0 +1,667 @@
+= Azure Storage
+The Azure cloud provider can be enabled with {productname} to allow Kubernetes pods to use Azure disks or Azure files as data volumes.
```
Comment (jenting) [inline adoc/admin-storage-azure.adoc:2]:
```suggestion
The Azure cloud provider can be enabled with {productname} to allow {kube} pods to use Azure disks or Azure files as persistent volumes.
```

File: adoc/admin-storage-azure.adoc [hunk 4]
```diff
@@ -0,0 +1,667 @@
...
+
+Please refer to link:{docurl}single-html/caasp-deployment/#bootstrap[Cluster Bootstrap] on how to setup Azure cloud provider enabled cluster.
+
+== Azure Disk
+
+[[create_azure_disk]]
+=== Create Disk
+. Install Azure CLI on local machine.
++
+For installation instructions, refer to link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install the Azure CLI].
```
Comment (jenting) [inline adoc/admin-storage-azure.adoc:14]: We have `zypper install azure-cli` ^^

File: adoc/deployment-bootstrap.adoc [hunk 5]
```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
...
+====
+The file `my-cluster/cloud/azure/azure.conf` must not be freely accessible.
+Please remember to set proper file permissions for it, for example `600`.
+====
+
+===== Example Azure Cloud Provider Configuration
+
+====
+    {
+        "cloud":"AzurePublicCloud",
```
Comment (jenting) [inline adoc/deployment-bootstrap.adoc:318]: nit
```suggestion
        "cloud": "AzurePublicCloud",
```

File: adoc/deployment-bootstrap.adoc [hunk 6]
```diff
@@ -290,6 +290,107 @@ these names will be used by the `AWS` cloud controller manager to reconcile node
...
+[[cluster.bootstrap.azure.create-service-principle]]
+===== Create Service Principle
+[IMPORTANT]
+====
+Ensure there is sufficient azure account permission. The account needs to be either a member of the `Owner` role, or have `Contributor` plus `User Access Administrator` roles.
+====
+
+. Install Azure CLI on local machine.
++
+For installation instructions, refer to link:https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest[Install the Azure CLI].
```
Comment (jenting) [inline adoc/deployment-bootstrap.adoc:365]: We have `zypper install azure-cli` ^^

Comment (innobead) [reply adoc/deployment-bootstrap.adoc:365]: cc @cclhsu

File: adoc/admin-storage-azure.adoc (hunk 4 above)
Comment (c3y1huang) [reply adoc/admin-storage-azure.adoc:14]: Link includes instructions how to install with zypper.

File: adoc/deployment-bootstrap.adoc (hunk 6 above)
Comment (c3y1huang) [reply adoc/deployment-bootstrap.adoc:363]: Link includes instructions how to install with zypper.

**Comment by c3y1huang** [comment]
This will wait until https://github.com/SUSE/doc-caasp/pull/804 gets merged to update parts relates to cluster provision.

**Comment by innobead** [comment]
cc @cclhsu

**Comment by c3y1huang** [comment]
rebased.

**Comment by c3y1huang** [comment]
rebased and fixed IDREF.

**Comment by r0ckarong** [comment]
@c3y1huang Added a commit with some minor integration, please rebase before pushing again.

**Comment by r0ckarong** [comment]
@c3y1huang Did anything change since the last update for this feature? Please have another review so I can merge this for the 4.5.2 release. Thank you!

**Comment by c3y1huang** [comment]
> @c3y1huang Did anything change since the last update for this feature? Please have another review so I can merge this for the 4.5.2 release. Thank you!

Nothing had changed since the last update. This one is ready for review and was just waiting for https://github.com/SUSE/doc-caasp/pull/878#issuecomment-647927137.

**Comment by c3y1huang** [comment]
rebased.

**Pull request merged**